### PR TITLE
Fix(#3239) NPE, for RegexFilter creator.

### PR DESCRIPTION
**Pull request by jaykataria1111**
fix for #3239 

## Checklist

* Base your changes on `2.x` branch if you are targeting Log4j 2; use `main` otherwise
* `./mvnw verify` succeeds (if it fails due to code formatting issues reported by Spotless, simply run `./mvnw spotless:apply` and retry)
* Non-trivial changes contain an entry file in the `src/changelog/.2.x.x` directory
* Tests for the changes are provided
* [Commits are signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (optional, but highly recommended)

### CODE REVIEW

**Review by jvz — APPROVED** [review]

File: log4j-core/src/main/java/org/apache/logging/log4j/core/filter/RegexFilter.java [hunk 1]
```diff
@@ -148,7 +148,8 @@ public static RegexFilter createFilter(
             LOGGER.error("A regular expression must be provided for RegexFilter");
             return null;
         }
-        return new RegexFilter(useRawMsg, Pattern.compile(regex, toPatternFlags(patternFlags)), match, mismatch);
+        return new RegexFilter(
+                useRawMsg != null && useRawMsg, Pattern.compile(regex, toPatternFlags(patternFlags)), match, mismatch);
```
Comment (ppkarwasz) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/filter/RegexFilter.java:152]: _Nit_: Maybe comparing `useRawMsg` to `Boolean.TRUE` might be simpler to read:
```suggestion
        return new RegexFilter(
                Boolean.TRUE.equals(useRawMsg), Pattern.compile(regex, toPatternFlags(patternFlags)), match, mismatch);
```

**Review by ppkarwasz — COMMENTED** [review]
@jaykataria1111,

Can you add a unit test to see if `createFilter` does not throw with all parameters (except `regex`) equal to `null`?

File: log4j-core/src/main/java/org/apache/logging/log4j/core/filter/RegexFilter.java (hunk 1 above)
Comment (jaykataria1111) [reply log4j-core/src/main/java/org/apache/logging/log4j/core/filter/RegexFilter.java:152]: +1, yeah I like the suggestion, let us do that!

**Comment by jaykataria1111** [comment]
@ppkarwasz added the test, and updated the code :)

**Comment by jaykataria1111** [comment]
@ppkarwasz , this one is still open I will rebase this just for tracking purposes, I do not recall any action items on this would you remember?

**Comment by jaykataria1111** [comment]
@ppkarwasz I have rebased the commit, let me know if anything else is needed on this or we can close this out. Thanks for this!

**Comment by jaykataria1111** [comment]
@ppkarwasz posted an update .

**Comment by ppkarwasz** [comment]
Thanks for your contribution!
